app/init.php

```diff
@@ -660,78 +661,23 @@ function (mixed $value) {
-Locale::setLanguageFromJSON('ne', __DIR__ . '/config/locale/translations/ne.json');
-Locale::setLanguageFromJSON('nl', __DIR__ . '/config/locale/translations/nl.json');
-Locale::setLanguageFromJSON('nn', __DIR__ . '/config/locale/translations/nn.json');
-Locale::setLanguageFromJSON('or', __DIR__ . '/config/locale/translations/or.json');
-Locale::setLanguageFromJSON('pa', __DIR__ . '/config/locale/translations/pa.json');
-Locale::setLanguageFromJSON('pl', __DIR__ . '/config/locale/translations/pl.json');
-Locale::setLanguageFromJSON('pt-br', __DIR__ . '/config/locale/translations/pt-br.json');
-Locale::setLanguageFromJSON('pt-pt', __DIR__ . '/config/locale/translations/pt-pt.json');
-Locale::setLanguageFromJSON('ro', __DIR__ . '/config/locale/translations/ro.json');
-Locale::setLanguageFromJSON('ru', __DIR__ . '/config/locale/translations/ru.json');
-Locale::setLanguageFromJSON('sa', __DIR__ . '/config/locale/translations/sa.json');
-Locale::setLanguageFromJSON('sd', __DIR__ . '/config/locale/translations/sd.json');
-Locale::setLanguageFromJSON('si', __DIR__ . '/config/locale/translations/si.json');
-Locale::setLanguageFromJSON('sk', __DIR__ . '/config/locale/translations/sk.json');
-Locale::setLanguageFromJSON('sl', __DIR__ . '/config/locale/translations/sl.json');
-Locale::setLanguageFromJSON('sn', __DIR__ . '/config/locale/translations/sn.json');
-Locale::setLanguageFromJSON('sq', __DIR__ . '/config/locale/translations/sq.json');
-Locale::setLanguageFromJSON('sv', __DIR__ . '/config/locale/translations/sv.json');
-Locale::setLanguageFromJSON('ta', __DIR__ . '/config/locale/translations/ta.json');
-Locale::setLanguageFromJSON('te', __DIR__ . '/config/locale/translations/te.json');
-Locale::setLanguageFromJSON('th', __DIR__ . '/config/locale/translations/th.json');
-Locale::setLanguageFromJSON('tl', __DIR__ . '/config/locale/translations/tl.json');
-Locale::setLanguageFromJSON('tr', __DIR__ . '/config/locale/translations/tr.json');
-Locale::setLanguageFromJSON('uk', __DIR__ . '/config/locale/translations/uk.json');
-Locale::setLanguageFromJSON('ur', __DIR__ . '/config/locale/translations/ur.json');
-Locale::setLanguageFromJSON('vi', __DIR__ . '/config/locale/translations/vi.json');
-Locale::setLanguageFromJSON('zh-cn', __DIR__ . '/config/locale/translations/zh-cn.json');
-Locale::setLanguageFromJSON('zh-tw', __DIR__ . '/config/locale/translations/zh-tw.json');
+
+$locales = Config::getParam('locale-codes', []);
+
+foreach ($locales as $locale) {
+    $code = $locale['code'];
+
+    $path = __DIR__ . '/config/locale/translations/' . $code . '.json';
+
+    if (!\file_exists($path)) {
+        $path = __DIR__ . '/config/locale/translations/' . \substr($code, 0, 2) . '.json'; // if `ar-ae` doesn't exist, look for `ar`
+        if (!\file_exists($path)) {
+            $path = __DIR__ . '/config/locale/translations/en.json'; // if none translation exists, use default from `en.json`
```

Comment:
Maybe we can add an exception here when something fails? This will help maintainers know and understand why a language isn't working

Reply:
added console error instead. Throwing would fail running instance as some translations don't exist for us as well.

Reply:
We need to add lots of json files, because we have quite a few locale missing?